### ISSUE TITLE
`TabView` assumes `Frame == Bounds`

### ISSUE THREAD

**Issue by tig** [gui-cs/Terminal.Gui#2990]
Similar to #2989, TabView does not work well if any of the `Frames` have any dimensions.

To repro, change the `TabViewExample.cs` scenario to give the TabView a border:

```cs
			tabView = new TabView () {
				X = 0,
				Y = 0,
				Width = 60,
				Height = 20,
				BorderStyle = LineStyle.Single
			};
```

![image](https://github.com/gui-cs/Terminal.Gui/assets/585482/d7b32f85-58d2-418b-893c-e6ab6d2acab8)

**Comment by BDisp** [comment]
With the `TabView` you can't use the border. You need to explicit draw the frame.

**Comment by tig** [comment]
> With the `TabView` you can't use the border. You need to explicit draw the frame.

That is precisely the bug I'm reporting. Border should work. So should margin and padding.

**Comment by BDisp** [comment]
Well I'm really curious to know if `WPF` and `Windows.Forms` use the margin or padding of the control to draw the tabs.

**Comment by tig** [comment]
So I'm clear: I'm not talking about the lines that make up the tabs themselves or the lines that make up the box beside and below the tabs (contiguous with the tabs). I'm talking about the space AROUND all of that. 

Am I making sense to you?

**Comment by BDisp** [comment]
Sorry @tig, I didn't realizes that you refer to the space around the `TabView`. I'll investigate that but I thing the `TabView` is overwriting something.

**Comment by BDisp** [comment]
Well the problem is `LineCanvas` of the `TabView` is used to deal with two borders. When I call the `DrawFrame` it add more lines to the `LineCanvas` , but seems they not touch each other. If it's then the `OnRenderLineCanvas` isn't work as supposed. Right?

**Comment by BDisp** [comment]
Here is how the `LineCanvas` looks like on `TabView`:

```
┌──────────────────────────────────────────────────────────┐
│                                                          │
│                                                          │
│┌────────────────────────────────────────────────────────┐│
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
││                                                        ││
│└────────────────────────────────────────────────────────┘│
└──────────────────────────────────────────────────────────┘
```

If I don't call the `OnRenderLineCanvas` on the `DrawFrame` the output is this:

![image](https://github.com/gui-cs/Terminal.Gui/assets/13117724/2387ed60-7b4b-43ca-87ea-d40270e6d367)

**Comment by BDisp** [comment]
> Regarding your changes I think we should be more elegant than _lines. But need to understand the use case

_lines is the field that already exist. I only made it public to test. What I was asking is if I can create a public property `Lines` to get/set this value.

```cs
var lc = LineCanvas.Clone();
```
`CS1061	'LineCanvas' does not contain a definition for 'Clone' and no accessible extension method 'Clone' accepting a first argument of type 'LineCanvas' could be found (are you missing a using directive or an assembly reference?)`

**Comment by BDisp** [comment]
No @tznind you have to test the code. if I save by reference, `lc` will be empty. The solution is the creation of `Line` property on the `LineCanvas` class. And the `Clone`  method doesn't exist.

```cs
var lc = LineCanvas;

DrawFrame (new Rect (0, startAtY, Bounds.Width,
	Math.Max (Bounds.Height - spaceAtBottom - startAtY, 0)), LineStyle.Single);

LineCanvas = lc;

```

**Comment by BDisp** [comment]
See #2980.

**Comment by BDisp** [comment]
This was already fixed on #2980.
